### DISCUSSION TITLE
[Question][Hw5] Should we consider unsigned comparison?

### DISCUSSION THREAD

**Discussion by nubi12336** [prosyslab-classroom/cs348-information-security#593] [Q&A]
In utils.ml, there are also unsigned comparison operators at Llvm.Icmp.
Do we have to consider 'unsigned' when we write 'cmp' or 'filter' part?
If so, should we cast signed integer to unsigned integer?

**Comment by Standchen** [comment]
In readme, it says:

> 2. Implementing Value Domains and Memory Domain
> 
> In this homework, you will implement the sign domain and the taint domain in the corresponding module. Students are required to implement sound abstract semantic functions declared in the modules. Assume that all integers are **signed** and ignore integer-overflow effects.

So I guess we can just ignore unsigned operators... maybe?

**Comment by KihongHeo** [comment]
Yes. Ignore unsigned. Test cases will only consider signed integers.
